cusub_cortex/state_machine/src/tasks/task.py

```diff
@@ -426,14 +444,31 @@ def set_new_time(self, seconds):
         self.timed_out = False
         if seconds != 0:
             self.timer = rospy.Timer(rospy.Duration(seconds), self.timer_callback)
+            self.smprint("Next task time: " + str(seconds) + "s")    
         else:
-            rospy.logwarn("No timeout monitoring on next task")
+            self.smprint("No timeout monitoring on next task", warn=True)
 
     def timer_callback(self, msg):
         self.timer.shutdown()
-        rospy.logerr("Task Timed Out")
+        self.smprint("Task Timed Out", warn=True)
         self.timed_out = True
 
     def timed_out(self):
         """ In objectives reference like: userdata.timeout_obj.timed_out """
-        return self.timed_out
\ No newline at end of file
+        return self.timed_out
+
+    def smprint(self, string, warn=False):
+        if warn:
+            rospy.loginfo("[" + bcolors.OKGREEN + self.name + bcolors.ENDC + "] " + bcolors.WARNING +"[WARN] "+ string + bcolors.ENDC)
+        else:
+            rospy.loginfo("[" + bcolors.OKGREEN + self.name + bcolors.ENDC + "] " + string)
+
```

Comment:
This is also defined 6 different times (the only difference seems to be "startup script"), and seems to only use the name part of the class, could it be defined in a separate static method that takes a name as a parameter, and then is imported by clients?

Reply:
Good catch yeah it gets defined everywhere that it couldn't be easily inherited... yeah doing some sort of importing thing might make that cleaner and more maintainable